### DISCUSSION TITLE
laravel 9.11 issues with mpdf

### DISCUSSION THREAD

**Discussion by wontone18** [mpdf/mpdf#1709] [General]
Hi

i am facing one issue installing the mpdf in laravel 9.11 version.

```
Problem 1
    - Root composer.json requires mpdf/mpdf ^8.1 -> satisfiable by mpdf/mpdf[v8.1.0, v8.1.1, v8.1.2].
    - mpdf/mpdf[v8.1.0, ..., v8.1.2] require psr/log ^1.0 || ^2.0 -> found psr/log[1.0.0, ..., 1.1.4, 2.0.0] but the package is fixed to 3.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require mpdf/mpdf:*" to figure out if any version is installable, or "composer require mpdf/mpdf:^2.1" if you know which you need.

```

i installed its dowgraded version composer require mpdf/mpd:* but that version giving me psrlog error

`Declaration of Mpdf\Mpdf::setLogger(Psr\Log\LoggerInterface $logger) must be compatible with Psr\Log\LoggerAwareInterface::setLogger(Psr\Log\LoggerInterface $logger): void`

**Comment by finwe** [comment]
mPDF does not support psr/log 3.x because of compatibility with lower PHP versions.
Find the `psr/log` line in your composer.json and change the version there to `^2.0`.
If there is no `psr/log` line in your composer.json, add it with version above and run composer update.
Then install mPDF as you did before.
If then there are still version mismatches, it is possible you will have to downgrade the library in your stack requiring `psr/log:^3.0`

**Reply by wontone18** [reply]
let me check it

**Comment by SaraKaterji** [comment]
I had the same problem yesterday, what I did is that I updated wamp server (to the latest) with a version that includes php8 and did a fresh laravel installation (also latest update) and then installed mpdf without any issues.
